### PR TITLE
Display conditional parameters properly in TrialTable

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

This PR solves [issue#734](https://github.com/optuna/optuna-dashboard/issues/734).

## What does this implement/fix? Explain your changes.

By this PR, TrialTable can:
1. display conditional parameters appropriately, and
2. filter missing value via filtering.

### CODE REVIEW

**Comment by c-bata** [comment]
@keisuke-umezawa Could you review this PR?

**Comment by c-bata** [comment]
Let me assign @porink0424 as an additional reviewer :pray:

**Comment by porink0424** [comment]
Sure.

**Review by porink0424 — APPROVED** [review]
LGTM!

**Comment by c-bata** [comment]
@keisuke-umezawa Please merge this PR after you approved 🙏

File: optuna_dashboard/ts/components/TrialTable.tsx
Comment (keisuke-umezawa) [inline optuna_dashboard/ts/components/TrialTable.tsx:109]: `isDynamicSpace` is calculated for each study, but `filterChoices` is set for each column. This `isDynamicSpace` should be implemented as the value whether the column is dynamic space.

```
import optuna


def objective(trial):
    category = trial.suggest_categorical("category", ["foo", "bar"])
    if category == "foo":
        return (trial.suggest_float("x1", 0, 10) - 2) ** 2
    else:
        return -((trial.suggest_float("x2", -10, 0) + 5) ** 2)


study = optuna.create_study(storage="sqlite:///test.db")
study.optimize(objective, n_trials=100)
```

In the above code, this study's `isDynamicSpace` is always true because some of columns are dynamic spaces. As a result,  `filterChoices` will be `filterChoicesWithNull` even though `category` is not dynamic space.

Comment (keisuke-umezawa) [reply optuna_dashboard/ts/components/TrialTable.tsx:109]: Or, something like this?

```
const filterChoices =
      s.distribution.type === "CategoricalDistribution"
        ? s.distribution.choices.map((c) => c.value)
        : undefined
values = trials[i].params.find((p) => p.name === s.name)?.param_external_value
if values.contain(null):
    filterChoices.append(null)
```

**Review by keisuke-umezawa — COMMENTED** [review]
Sorry for being late. I left one comment. Could you check it?

**Comment by nabenabe0928** [comment]
@keisuke-umezawa Thank you for the suggestion, I addressed your point!

**Review by keisuke-umezawa — APPROVED** [review]
LGTM! Thank you for quick fix.

**Pull request merged**
